ui/jobs/components/index.ts

```diff
@@ -214,6 +214,7 @@ export class ComponentManager {
 
       // initialize components
       this.component = this.getJobComponents(job);
+      this.component.reset();
```

Comment:
I'm not sure I understand this.  This function returns a constructed object.  If those objects are not valid, we should fix it in the constructors of those objects.

Reply:
But this way we don't need to duplicate those codes in the reset function in every constructor, so I think putting it here is good.

Reply:
What about calling `this.reset()` in the constructor then? This is an abstract thought, but I feel that in general if you create an object it should be initialized, because that way it can't ever be used incorrectly.

Reply:
But then you should write this.reset() in every components...

Reply:
Yes, but I think repeating is less bad? I think it's just wrong in general to have a class where calling `new` leaves you with a half-finished object.  I think it's just a bad practice, because then somebody in the future can call `new ThingComponent` and then be surprised that they have to do an extra step.

Reply:
Oh, I see your point. but I think it is better to initialize those values in the "box"-es itself instead. Maybe in another PRs.
Sorry I am a bit busy these days, and not react immediately, I'll tweak some more things and then can be merged.

Reply:
No worries! I know everybody's busy these days.  :)

I'm not sure I understand what you mean by "box"-es here, but I'll take a look when you tweak some things.